addons/mail/static/src/models/notification/notification.js

```diff
@@ -44,6 +44,46 @@ function factory(dependencies) {
             return data2;
         }
 
+        /**
+         * @private
+         * @returns {boolean}
+         */
+        _computeIsFailure() {
+            return ['exception', 'bounce'].includes(this.notification_status);
+        }
+
+        /**
+         * @private
+         * @returns {boolean|FieldCommand}
+         */
+        _computeIsFromCurrentUser() {
+            if (!this.messaging || !this.messaging.currentPartner || !this.message || !this.message.author) {
+                return clear();
+            }
+            return this.messaging.currentPartner === this.message.author;
+        }
+
+        /**
+         * @private
+         * @returns {FieldCommand}
+         */
+        _computeNotificationGroup() {
+            if (!this.isFailure || !this.isFromCurrentUser) {
+                return clear();
+            }
+            const thread = this.message.originThread;
+            // Notifications are grouped by model and notification_type.
+            // Except for channel where they are also grouped by id because
+            // we want to open the actual channel in discuss or chat window
+            // and not its kanban/list/form view.
+            return insertAndReplace({
+                notification_type: this.notification_type,
+                res_id: thread.model === 'mail.channel' ? thread.id : null,
```

Comment:
It's not great to deal with `null`... It's handy work-around for setting it with how identifying fields are defined, but it's not pretty...

I think future plan is to have a model for a (python) model and a model for a (python) record, so the modelling clearer without needing workaround like `null`. Right now it's too problematic to have this. so it's fine `null` for the time being as long as we are aware of this technical debt.

Reply:
But even if we have python model, here the goal of the group is precisely to group multiple threads, not to have just one.

It would already be possible now if it wasn't for that part.

The problem is that we would need complex identifying field such as "model of all threads" with "guarantee that all threads have the same model".

Reply:
Yeah, it's not trivial. I'm good with `null` workaround for the moment. No explicit call to compute method is a big improvement.